src/parsers/manifestjson.js

```diff
@@ -70,6 +70,12 @@ export default class ManifestJSONParser extends JSONParser {
       };
     } else {
       // We've parsed the JSON; now we can validate the manifest.
+      if (
```

Comment:
@eviljeff we should move this into `_validate`, after the manifest has been validated using the schemas imported from Firefox, otherwise every schema validation error/warning related to the `browser_specific_settings` will be logged twice: once for `browser_specific_settings` and then again for `applications` ( because we have already overwritten "applications" with the content of "browser_specific_settings").

If you prefer I can apply this change for you, or you may apply the following diff and update this PR and I'll squash and merge it right away: 

```
diff --git a/src/parsers/manifestjson.js b/src/parsers/manifestjson.js
index 6314f6e..30636cb 100644
--- a/src/parsers/manifestjson.js
+++ b/src/parsers/manifestjson.js
@@ -70,12 +70,6 @@ export default class ManifestJSONParser extends JSONParser {
       };
     } else {
       // We've parsed the JSON; now we can validate the manifest.
-      if (
-        this.parsedJSON.browser_specific_settings &&
-        this.parsedJSON.browser_specific_settings.gecko
-      ) {
-        this.parsedJSON.applications = this.parsedJSON.browser_specific_settings;
-      }
       this.selfHosted = selfHosted;
       this.isLanguagePack = Object.prototype.hasOwnProperty.call(
         this.parsedJSON,
@@ -174,6 +168,13 @@ export default class ManifestJSONParser extends JSONParser {
       });
     }
 
+    if (
+      this.parsedJSON.browser_specific_settings &&
+        this.parsedJSON.browser_specific_settings.gecko
+    ) {
+      this.parsedJSON.applications = this.parsedJSON.browser_specific_settings;
+    }
+
     if (this.parsedJSON.content_security_policy) {
       this.validateCspPolicy(this.parsedJSON.content_security_policy);
     }
```